### PR TITLE
Add tab scrolling support for dialog windows

### PR DESCRIPTION
**Pull request by tamplan**

### CODE REVIEW

**Comment by raveit65** [comment]
@tamplan 
I can merge only the git.mk commit locally.
The other one needs to be reviewed more.
Are you able to rebase this PR after this particular merge?
Or does this brings chaos in your worflow :)

**Comment by tamplan** [comment]
> @tamplan
> I can merge only the git.mk commit locally.
> The other one needs to be reviewed more.
> Are you able to rebase this PR after this particular merge?
> Or does this open the chaos in your workflow :)

I will separate the two commits one per a specific branch, I will update this soon.

**Comment by tamplan** [comment]
> @tamplan
> I can merge only the git.mk commit locally.
> The other one needs to be reviewed more.
> Are you able to rebase this PR after this particular merge?
> Or does this open the chaos in your workflow :)

I will separate the two commits one per a specific branch, I will update this soon.

> @tamplan
> I can merge only the git.mk commit locally.
> The other one needs to be reviewed more.
> Are you able to rebase this PR after this particular merge?
> Or does this open the chaos in your workflow :)

I have indicate yet i am not using the good method in my workflow.
Let's me explain how i do.
First, i'm not a Master of Git i search a tutorial like this on https://www.fotozik.fr/github-mettre-a-jour-un-fork (sorry in French).
After repositories set (local, my fork and remote with upstream here) I use:

- git checkout master && git fetch upstream
- git rebase upstream/master (here, it's perhaps the problem)
- git push origin master

to have the forked and local repository always updated (after a PR i purpose or another one is merged).

"git rebase upstream/master" is used to not keep trace of merge in historial i have well understood the tutorial.
The different commits (originally done on specific branch plus a PR for each subject, with PR approved but not merged) are all mixed on the last PR like here i presume.
Imagine when i forgot to delete the branch in the forked repository too :)

Did someone give me the tips to do the right way?

Thanks by advance.

Regards

**Comment by tamplan** [comment]
Add tab scrolling support for GTK3 is PR272 mixed by error with this 273.

**Comment by raveit65** [comment]
Did you ever add upstream (we)repo to your local repo?

Exxample: sync with upstream
```
git remote add upstream git://github.com/user/repo.git    // this will add and upstream repo to your local repo, replace user/repo with mate-desktop/eom

git pull upstream master   // this pulls latest code from upstream master branch into your master branch, Take care on which branch you are!!!!!

git push origin    // this push new code from your local master to your github repo (master)
```

After this switch to your branch which you are using for the pull request, and rebase the branch with master
`git rebase master`
Than update you github PR branch with you local PR branch with
`git push -f`
(-f) = --force, Be carefull, this can't be undo.
Now the PR is update automatic.

**Comment by tamplan** [comment]
> Did you ever add upstream (we)repo to your local repo?
> 
> Example: sync with upstream
> 
> ```
> git remote add upstream git://github.com/user/repo.git    // this will add and upstream repo to your local repo, replace user/repo with mate-desktop/eom
> 
> git pull upstream master   // this pulls latest code from upstream master branch into your master branch, Take care on which branch you are!!!!!
> 
> git push origin    // this push new code from your local master to your github repo (master)
> ```
> 
> After this switch to your branch which you are using for the pull request, and rebase the branch with master
> `git rebase master`
> Than update you github PR branch with you local PR branch with
> `git push -f`
> (-f) = --force, Be carefull, this can't be undo.
> Now the PR is update automatic.

Normally repositories settings is good like in the tutorial i give the link.
I will read several times and translate your answer to be sure and i understand what i need to do or not.

Thanks

**Comment by raveit65** [comment]
I just picked `add git.mk to generate .gitignore` to master
https://github.com/mate-desktop/eom/commit/8ef04724513572bf5ef22a8ab6ea8d07350d9851

**Comment by raveit65** [comment]
Branch has no conflicts and can be merged. So leave everything like it is, do nothing and wait for another reviewer.

**Comment by tamplan** [comment]
> Branch has no conflicts and can be merged. So leave everything like it is, do nothing and wait for another reviewer.

Ok, I leave all (272 and 273) like there are, awaiting merging after approval.

Thanks

**Comment by tamplan** [comment]
> I have only reviewed the changes that are related to the change of page of the notebook by using the mouse.
> 
> Test: Image Properties dialog requires an image with exif data
> https://github.com/ianare/exif-samples/blob/master/jpg/Canon_40D.jpg
> It works fine for me.
> 
> Test: Eye of Mate Preferences dialog
> It works fine for me.
> 
> From my point of view the name of the function `eom_util_dialog_page_scroll_event_cb` can be improved a bit, maybe just adding the word notebook and shortening it a little.

Agree with you about the name of the function. Changed perhaps later after the merge because of the PR mixed i purpose.

**Comment by rbuj** [comment]
I reviewed this at https://github.com/mate-desktop/eom/pull/272

**Comment by raveit65** [comment]
> We reviewed this at #272

LOL, and author closed PR with 2 approvals

@tamplan 
Using git for editing commits, rebasing or using the right branch for PRs are mandatory skills.
And i gave you a tutorial about how to use the `git rebase -i` command in another PR a while ago.
But we can help this time.
Please rename the function with another commit and add it to this PR.
We can squash the commit into the first one when merging.

**Comment by tamplan** [comment]
I use:

- `pluma_preferences_dialog_page_scroll_event_cb` in https://github.com/mate-desktop/pluma/commit/a31a8facfba9693f7337631dfa305c64a5663e2e
- `gsm_properties_dialog_page_scroll_event_cb` in https://github.com/mate-desktop/mate-session-manager/commit/454dccb8138d611da9798bde106363fa9cf83e01

I purpose `eom_notebook_page_scroll_event_cb`, are you ok with this?

**Comment by tamplan** [comment]
Function name changed, i think is correct now.

File: src/eom-util.c [hunk 1]
```diff
@@ -427,3 +427,47 @@ eom_util_show_file_in_filemanager (GFile *file, GtkWindow *toplevel)
...
+        case GDK_SCROLL_RIGHT:
+        case GDK_SCROLL_DOWN:
+            gtk_notebook_next_page (notebook);
+            break;
+        case GDK_SCROLL_LEFT:
+        case GDK_SCROLL_UP:
+            gtk_notebook_prev_page (notebook);
+            break;
+        case GDK_SCROLL_SMOOTH:
+        break;
```
Comment (rbuj) [inline src/eom-util.c:469]: You can also handle  GDK_SCROLL_SMOOTH.
See https://github.com/mate-desktop/mate-terminal/commit/622e2d1272bf2d7a99d147482ae6a51b0a77f188

Comment (tamplan) [reply src/eom-util.c:469]: Yes, i have seen your other PR about this subject.
There is another PR i purpose about notebook scroll event in the mate-system-monitor.
I'll follow your code to update it.

Comment (tamplan) [reply src/eom-util.c]: Done.

**Comment by raveit65** [comment]
@rbuj 
Any concerns left or can we squash+merge it?
For some reasons we have 2 commits here.

File: src/eom-util.c (hunk 1 above)
Comment (rbuj) [reply src/eom-util.c]: Do not mark conversations as resolved.

**Comment by raveit65** [comment]
Resulting commit is merged here https://github.com/mate-desktop/eom/commit/c331ec988f543ddfd5e0636d15f7aeafeefa1b0f

File: src/eom-util.c (hunk 1 above)
Comment (rbuj) [reply src/eom-util.c]: why didn't you use the same code to handle GDK_SCROLL_SMOOTH direction?

Comment (rbuj) [reply src/eom-util.c]: code indent will be fixed on #277

**Pull request merged**